### ISSUE TITLE
Unexpected error for webextension submission

### ISSUE THREAD

**Issue by vasilicamihasca** [mozilla/addons-server#2699]
**Steps to reproduce:**
1. Try to submit the attached web-extension on AMO-dev or AMO-stage.

**Expected results:**
The webextension is successfully submitted.

**Actual results:**
The web-extension cannot be submitted because of an unexpected error.

**Additional notes:**
- This issue reproduces on AMO-dev and AMO-stage using Firefox 49.0a1 (2016-05-17) under Windows 10 64-bit.
- I did not encountered this error while submitting on AMO-production.
- Please see the attached screenshot for this issue:
![2016-05-18_1541](https://cloud.githubusercontent.com/assets/18114210/15359288/4663f52c-1d11-11e6-8510-8151e0cf7d6c.png)
![2016-05-18_1541_001](https://cloud.githubusercontent.com/assets/18114210/15359291/4b52653c-1d11-11e6-91f9-14493e3d365b.png)
- I am attaching the web-extensions for which I’ve got these errors:
[Thesaurus--Synonym-4-Right-Click.zip](https://github.com/mozilla/addons-server/files/270516/Thesaurus--Synonym-4-Right-Click.zip)

**Comment by andymckay** [comment]
```
<--- Last few GCs --->

   14255 ms: Mark-sweep 1366.7 (1423.4) -> 1252.2 (1309.0) MB, 1044.5 / 0 ms [allocation failure] [GC in old space requested].
   15311 ms: Mark-sweep 1252.2 (1309.0) -> 1252.2 (1309.0) MB, 1055.2 / 0 ms [allocation failure] [GC in old space requested].
   16389 ms: Mark-sweep 1252.2 (1309.0) -> 1252.1 (1308.0) MB, 1078.2 / 0 ms [last resort gc].
   17495 ms: Mark-sweep 1252.1 (1308.0) -> 1252.1 (1308.0) MB, 1106.1 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3175861c9fa9 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [/Users/andy/sandboxes/addons-linter/node_modules/eslint/lib/token-store.js:~11] [pc=0x19c878873c39] (this=0x317586104189 <undefined>,tokens=0xe9a87efcc79 <JS Array[2189797]>)
    2: new constructor(aka SourceCode) [/Users/andy/sandboxes/addons-linter/node_modules/eslint/lib/util/source-code.js:126] [pc=0x19c87876693e] (this=0xd3a5e8dca71 <a SourceCod...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
[1]    4077 abort      ~/sandboxes/addons-linter/bin/addons-linter .
```
It's probably on this:
```
-rw-r--r--   1 andy  wheel    18M  6 Oct  2014 background.js
```

**Comment by andymckay** [comment]
There's a ginormous list in background.js. Don't consider this a blocker.

**Comment by muffinresearch** [comment]
Locally I can run this but it take forever and uses ~1.85gb of memory but it ran successfully. 

Default memory allocation for node is 1gb on 64bit and 512mb on 32bit. 

Yesterday eslint v1.9 ran successfully locally but it also uses a similar amount of memory and takes minutes to run at a guess I'd say 8+ minutes.

Upgrading to eslint latest (v2.10.2)  I reproduced the same out of memory error just running eslint against `background.js`.

**Comment by muffinresearch** [comment]
Installing latest eslint globally using node 0.12.7 via nvm gave this result:

```
$ time eslint ~/Downloads/Thesaurus--Synonym-4-Right-Click/background.js

real	1m0.071s
user	1m6.780s
sys	0m1.254s
```

**Comment by muffinresearch** [comment]
And on the same node version:

```
$ time bin/addons-linter ~/Downloads/Thesaurus--Synonym-4-Right-Click.xpi
{"count":0,"summary":{"errors":0,"notices":0,"warnings":0},"metadata":{"id":"SynonymRightClickIcatest@mozilla.com","manifestVersion":2,"name":"Synonym 4 Right Click Ica Test","type":1,"version":"2.0.3","architecture":"extension","emptyFiles":[],"jsLibs":{}},"errors":[],"notices":[],"warnings":[]}

real	1m47.141s
user	2m0.011s
sys	0m1.551s
```

**Comment by cosminbadescuSV** [comment]
I receive the same error to this webextension: [Kifi.zip](https://github.com/mozilla/addons-server/files/274563/Kifi.zip), should I file a new bug?

**Comment by andymckay** [comment]
That's a different error, please file a new bug, on addons-linter:

```
tsty ~/sandboxes/addons-linter/bin/addons-linter . --stack                     
Error: CSS url() "initial" is invalid
    at extractCSSUri (/Users/andy/sandboxes/addons-linter/dist/webpack:/src/utils.js:223:11)
    at isLocalCSSUri (/Users/andy/sandboxes/addons-linter/dist/webpack:/src/utils.js:230:21)
    at Object.detectBadMozBindingURL (/Users/andy/sandboxes/addons-linter/dist/webpack:/src/rules/css/detectBadMozBinding.js:11:13)
    at CSSScanner.processCode (/Users/andy/sandboxes/addons-linter/dist/webpack:/src/scanners/css.js:50:7)
    at /Users/andy/sandboxes/addons-linter/dist/webpack:/src/scanners/css.js:64:22
    at run (/Users/andy/sandboxes/addons-linter/node_modules/core-js/modules/es6.promise.js:89:22)
    at /Users/andy/sandboxes/addons-linter/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/Users/andy/sandboxes/addons-linter/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by tofumatt** [comment]
So this isn't an MVP anymore I think; the file is quite large. As I said on the call; the purpose of the addons-linter is to not allow bad code through. I would classify this as "bad code".

**Comment by andymckay** [comment]
Fair enough, is there anything we need to upstream? File a bug or just see if it improves?

**Comment by tofumatt** [comment]
We could file a bug; I’ll keep this assigned to me to remember to do that 👍🏻

**Comment by muffinresearch** [comment]
I think the actionable item here would be to flag files over a certain size without getting into parsing them at all.

**Comment by vasilicamihasca** [comment]
I’m not quite sure which is the intended behavior here while installing the attached webextension from description. 
Tested on Firefox 50.0a1 on AMO-dev and I received the following error: “Your add-on failed validation with 1 error. Sorry, we couldn't load your WebExtension” 
I’m also attaching screenshots with this error: 
![2016-06-08_1023](https://cloud.githubusercontent.com/assets/18114210/15890583/fdce93a8-2d78-11e6-885d-a699d1f8ca20.png)
![2016-06-08_1301](https://cloud.githubusercontent.com/assets/18114210/15890627/3137a6a8-2d79-11e6-962a-5c2b8b55581c.png)

@tofumat Can you please confirm that this is the correct behaviour?

**Comment by tofumatt** [comment]
@vasilicamihasca That is not the correct behaviour, but it isn't actually fixed because we don't have a release for that fix yet on addons-server.

Give me about thirty minutes and I can push a new version that fixes it, then it can be tested.

It should give a specific error about a file bring too large.

**Comment by tofumatt** [comment]
This is now ready for testing on -dev

**Comment by tofumatt** [comment]
Because @wagnerand wanted new hashes in this will be on stage today.

Could someone from QA test on -dev and/or ping me on IRC and I'll let you know when it's ready for stage testing? Thanks.

**Comment by vasilicamihasca** [comment]
Verified fixed on AMO-dev using Firefox 50.0a1 (2016-06-09). The proper error message is successfully displayed.
Postfix screenshot:
![2016-06-09_1814](https://cloud.githubusercontent.com/assets/18114210/15935195/23c63c18-2e6f-11e6-93ef-33ffb2bd2635.png)
